lib/bojangles.rb

```diff
@@ -87,25 +87,36 @@ def cached_error_messages
     end
   end
 
+  # Only the first two lines of an error message need to match for us to decide
+  # that we've already sent it.
+  def compare_errors(current_errors, old_errors)
+    new_errors = current_errors.reject do |error|
+      old_errors.any? { |old| error.lines.first(2) == old.lines.first(2) }
+    end
+    resolved_errors = old_errors.reject do |error|
+      current_errors.any? { |new| error.lines.first(2) == new.lines.first(2) }
```

Comment:
There's no `new` method on modules, so this is the library they use for syntax highlighting, yes.